add basic usage example to demos on docs pages

falkben opened the pull request.

falkben committed:
add basic usage example to demos on docs pages

falkben committed:
remove the "comparisons" header

MrAE reviewed (approved):
Nice and with Gigantum!

falkben committed:
update Python requirements

update link to gigantum notebook from each notebook

add link to rerf gigantum on sidebar

falkben committed:
Merge branch 'staging' of https://github.com/neurodata/RerF into ben/example_usage_nb
